src/containerapp/azext_containerapp/_validators.py

```diff
@@ -6,21 +6,33 @@
 
 import re
 from azure.cli.core.azclierror import (ValidationError, ResourceNotFoundError, InvalidArgumentValueError,
-                                       MutuallyExclusiveArgumentError)
+                                       MutuallyExclusiveArgumentError, RequiredArgumentMissingError)
 from msrestazure.tools import is_valid_resource_id
 from knack.log import get_logger
 
 from ._clients import ContainerAppClient
 from ._ssh_utils import ping_container_app
 from ._utils import safe_get, is_registry_msi_system
-from ._constants import ACR_IMAGE_SUFFIX, LOG_TYPE_SYSTEM
+from ._constants import ACR_IMAGE_SUFFIX, LOG_TYPE_SYSTEM, MAXIMUM_SECRET_LENGTH
+from urllib.parse import urlparse
 
 
 logger = get_logger(__name__)
 
 
 # called directly from custom method bc otherwise it disrupts the --environment auto RID functionality
-def validate_create(registry_identity, registry_pass, registry_user, registry_server, no_wait):
+def validate_create(registry_identity, registry_pass, registry_user, registry_server, no_wait, source=None, repo=None):
+    if source and repo:
+        raise MutuallyExclusiveArgumentError("Cannot use --source and --repo together. Can either deploy from a local directory or a GitHub repository")
+    if source or repo:
+        if not registry_server or not registry_user or not registry_pass:
+            raise RequiredArgumentMissingError('Usage error: --registry-server, --registry-username and --registry-password are required while using --source or --repo.')
+    if repo and registry_server and "azurecr.io" in registry_server:
+        parsed = urlparse(registry_server)
```

Comment:
@cormacpayne we should check this condition for source also?

Reply:
@snehapar9 we should since we have the condition above to validate that `registry_server` is provided when `source` is provided -- we should be able to indent this entire block and remove the `repo` and `registry_server` checks to make it straightforward:

```python
if source or repo:
  if not registry_server or not registry_user or not registry_pass:
    raise RequiredArgumentMissingError('Usage error: --registry-server, --registry-username and --registry-password are required while using --source or --repo.')
  if "azurecr.io" in registry_server:
    parsed = urlparse(registry_server)
    registry_name = (parsed.netloc if parsed.scheme else parsed.path).split(".")[0]
    if registry_name and len(registry_name) > MAXIMUM_SECRET_LENGTH:
        raise ValidationError(f"--registry-server ACR name must be less than {MAXIMUM_SECRET_LENGTH} "
                              "characters when using --repo")```